src/crypto.c

```diff
@@ -541,8 +545,18 @@ void generate_keypair(Keypair *keypair, const uint32_t account)
     os_perso_derive_node_bip32(CX_CURVE_256K1, bip32_path, BIP32_PATH_LEN, keypair->priv, NULL);
     scalar_from_bytes(keypair->priv);
 
-    // Generate public key
-    generate_pubkey(&keypair->pub, keypair->priv);
+    // Checking cached_keypair.pub is not NULL is a workaround for the linker.
+    // Current linker script does not allow .data section to be non-empty (last_account can't be initialized).
+    if (account == last_account && memcmp(&cached_keypair.pub, &aff_zero, sizeof(Affine)) != 0) {
+        // Private key can't be cached for security reasons, so it is always computed (deterministically)
```

Comment:
It is not the ideal path, but we could encrypt the last private key and keep it on memory. We have encrypt data in other apps, and it is fast and deterministic. I am not quite sure about the idea, but is is a possibility

Reply:
I had tried caching the privkey in plaintext for testing performance, exactly with this idea in mind in case there had been any improvement. What I saw is that app's performance is only affected by the pubkey computation (At least to the user's eye).
Do you think it's worth a shot? Probably through more accurate benchmarking we can see results.

Reply:
If you already tested it, and there is no much performance improvement, then I think it is ok. It was mostly a thought, but I am ok with it if the improvement is already there caching the pub key.